custom_components/localtuya/common.py

```diff
@@ -101,30 +118,32 @@ def async_config_entry_by_device_id(hass, device_id):
     """Look up config entry by device id."""
     current_entries = hass.config_entries.async_entries(DOMAIN)
     for entry in current_entries:
-        if entry.data[CONF_DEVICE_ID] == device_id:
+        if device_id in entry.data[CONF_DEVICES]:
```

Comment:
After upgrade; in my logs:
```
Logger: homeassistant
Source: custom_components/localtuya/common.py:121
Integration: LocalTuya ([documentation](https://github.com/rospogrigio/localtuya/), [issues](https://github.com/rospogrigio/localtuya/issues))
First occurred: 9:06:13 PM (7 occurrences)
Last logged: 9:06:15 PM

Error doing job: Exception in callback _SelectorDatagramTransport._read_ready()
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 1029, in _read_ready
    self._protocol.datagram_received(data, addr)
  File "/config/custom_components/localtuya/discovery.py", line 70, in datagram_received
    self.device_found(decoded)
  File "/config/custom_components/localtuya/discovery.py", line 79, in device_found
    self._callback(device)
  File "/config/custom_components/localtuya/__init__.py", line 103, in _device_discovered
    entry = async_config_entry_by_device_id(hass, device_id)
  File "/config/custom_components/localtuya/common.py", line 121, in async_config_entry_by_device_id
    if device_id in entry.data[CONF_DEVICES]:
KeyError: 'devices'
```

Unclear exactly what's happening / may be a mistake on my part.

Reply:
This is strange because your config entry should have been migrated and after the migration that key should exist. Maybe there's a race condition with the discovery? I'll check this.